### PR TITLE
Rename the LegIntermediates element to LegIntermediate (and add an example for ParallelService from ÖBB)

### PR DESCRIPTION
**Pull request by ue71603**
working on https://github.com/VDVde/OJP/issues/398 Create TR_ParallelService_OeBB.xml

Als fix LegIntermediates -> LegIntermediate

### CODE REVIEW

**Comment by ue71603** [comment]
An open question during review: Should the "main" service be copied to parallel service as well (to tell that NJ 490 is only NJ 490 between stations 1 to 5). Otherwise we can't say that.

**Comment by trurlurl** [comment]
Which case should be expressed?

Case (1)
1
490 | 40490
5
590 | 50490
7
590
12

Case (2)
1
490 | 40490
5
490 | 590 | 50490
7
490 | 590
12

My interpretation might be too intuitive or simplistic, but I would expect that the Service element always applies to the complete parent which should be a dated journey. That would mean case (2), and a ParallelService 1-490-5 would not make any sense.  

To represent case (1) we'd need two dated journeys, one for Austria (Service 490, ParallelService 40490) and one for Germany (Service 590, ParallelService 50490 until Nürnberg). 

Perhaps I miss some complexities?

**Comment by ue71603** [comment]
What really 

> Which case should be expressed?
> 
> Case (1) 1 490 | 40490 5 590 | 50490 7 590 12
> 
> Case (2) 1 490 | 40490 5 490 | 590 | 50490 7 490 | 590 12
> 
> My interpretation might be too intuitive or simplistic, but I would expect that the Service element always applies to the complete parent which should be a dated journey. That would mean case (2), and a ParallelService 1-490-5 would not make any sense.
> 
> To represent case (1) we'd need two dated journeys, one for Austria (Service 490, ParallelService 40490) and one for Germany (Service 590, ParallelService 50490 until Nürnberg).
> 
> Perhaps I miss some complexities?

@trurlurl : What really happens on the rail is use case (1).   However we need to provide some basic information for the Service (even when it is only going from stop 1-7. And I think departure is the important place.  And really in (1) we shorten things and are not quite "true". However it is simpler. It would be always possible to model as two TimedLeg with a TransferLeg with 0 minutes and remainInVehicle. In many cases this would not be helpful. That's why I propose a duplication of the  490 1- 5 Service

**Comment by herlitze** [comment]
I would suggest splitting the line at "some station" and have a TransferLeg with TransferMode remainInVehicle between the two Legs. The result would be:

TimedLeg
Wien -> some station   - NJ 590 -> (Hamburg) - NJ50490 -> (Amsterdam)
NJ 490/NJ40490 

TransferLeg
some station -> some station
with TransferMode remainInVehicle

TimedLeg
some station -> Hamburg
NJ 590

...

I know this is not the prettiest of all solutions, but I dislike the idea of having the "main" service copied to parallel service as well.

**Comment by ue71603** [comment]
@skinkie  @Aurige : Having a split is ok for me as the train number really changes. What do you think? If ok for you, then I will do as @herlitze  suggests.

**Comment by ue71603** [comment]
@trurlurl  @skinkie  discussed with OeBB. If ok, then we can merge it.

**Review by trurlurl — DISMISSED** [review]

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+									<siri:StopPointRef>spp:1</siri:StopPointRef>
+									<StopPointName>
+										<Text xml:lang="de">Wien Hbf</Text>
+									</StopPointName>
+									<ServiceDeparture>
+										<TimetabledTime>2023-08-15T20:13:00Z</TimetabledTime>
+									</ServiceDeparture>
+									<Order>1</Order>
+								</LegBoard>
+								<LegIntermediates>
```
[the remark this answers is not in the document]
Comment (ue71603) [reply examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml]: changed

**Comment by ue71603** [comment]
@trurlurl @skinkie : Can we do this one next?

**Review by skinkie — DISMISSED** [review]

**Review by sgrossberndt — APPROVED** [review]

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml [hunk 2]
```diff
@@ -0,0 +1,243 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<OJP xmlns="http://www.vdv.de/ojp" xmlns:siri="http://www.siri.org.uk/siri" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.1-dev" xsi:schemaLocation="http://www.vdv.de/ojp ../../../OJP.xsd">
+	<!--
+We have a parallel train NJ 490 and NJ 40490
+Changement of NJ 490 to NJ590 at the border to Germany. This should be modeled with ParallelService
```
Comment (sgrossberndt) [inline examples/functions/trip/TR_ParallelService_OeBB_wrong.xml:5]: Please add a description here as well why this is "wrong"

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+				<CalcTime>762</CalcTime>
+				<TripResult>
+					<Id>ID-48D1A6A7-2659-4433-9795-F0A0840A4938</Id>
+					<Trip>
+						<Id>ID-48D1A6A7-2659-4433-9795-F0A0840A4938</Id>
+						<Duration>PT48M</Duration>
+						<StartTime>2023-08-15T18:08:00Z</StartTime>
+						<EndTime>2023-08-15T18:56:00Z</EndTime>
+						<Transfers>1</Transfers>
+						<!-- is one even when it is a remain seated -->
```
Comment (sgrossberndt) [inline examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml:26]: What does "is one even when it is a remain seated" mean? Is a parallel train? Trip?

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+									<TrainNumber>1231012</TrainNumber>
+									<siri:VehicleRef>B</siri:VehicleRef>
+									<DestinationText>
+										<Text>Amsterdam</Text>
+									</DestinationText>
+								</ParallelService>
+							</TimedLeg>
+						</Leg>
+						<Leg>
+							<!-- the TransferLeg with remaininVehicle -->
```
Comment (sgrossberndt) [inline examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml:139]: remainInVehicle

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+										<Name>
+											<Text xml:lang="de">Zug</Text>
+										</Name>
+										<ShortName>
+											<Text xml:lang="de">NJ</Text>
+										</ShortName>
+									</Mode>
+									<PublishedServiceName>
+										<Text xml:lang="de">NJ 490</Text>
+										<!-- the PublishedService changes in the middle, therefore having this PublishedServiceName and the train number
```
Comment (sgrossberndt) [inline examples/functions/trip/TR_ParallelService_OeBB_wrong.xml:146]: What does "the PublishedService changes in the middle" mean?

**Review by skinkie — APPROVED** [review]

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+										<Text>NJ 590</Text>
+									</PublishedServiceName>
+									<TrainNumber>123123</TrainNumber>
+									<siri:VehicleRef>A</siri:VehicleRef>
+									<DestinationText>
+										<Text>Hamburg Altona</Text>
+									</DestinationText>
+								</ParallelService>
+								<ParallelService>
+									<!-- the German part of the train to Amsterdam, only to Nürberg afterwards separated -->
```
Comment (sgrossberndt) [inline examples/functions/trip/TR_ParallelService_OeBB_wrong.xml:219]: What does "only to Nürberg afterwards separated" mean? Do you mean "only to Nürnberg after being separated"?

**Review by sgrossberndt — CHANGES_REQUESTED** [review]

**Comment by sgrossberndt** [comment]
In my opinion "Rename the LegIntermediates element to LegIntermediate" should be separated from "Add an example for ParallelService from ÖBB". Do one thing at a time.

**Review by trurlurl — DISMISSED** [review]

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml (hunk 3 above)
Comment (ue71603) [reply examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml:26]: @herlitze @normanoffel   This is a documentation question, we need to address: Is Transfers the number of TransferLegs or is it 0 in the case of a TransferType remainInVehicle?

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml (hunk 2 above)
Comment (ue71603) [inline examples/functions/trip/TR_ParallelService_OeBB_wrong.xml:5]:
```suggestion
Changement of NJ 490 to NJ590 at the border to Germany. This should be modeled with ParallelService.

This example is not modeled in a good way as the Service (NJ490) is not really going all the way. Also it needed  to be duplicated into ParallelService, where it is accuractely modeled. It is always better to model such a constellation by using two TimedLegs with a TransferLeg with the TransferType "remainInVehicle"
```

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml (hunk 3 above)
Comment (ue71603) [reply examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml:26]: When I have your answer, I adapt the PR

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml (hunk 4 above)
Comment (ue71603) [inline examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml:139]:
```suggestion
							<!-- the TransferLeg with remainInVehicle -->
```

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml (hunk 2 above)
Comment (ue71603) [reply examples/functions/trip/TR_ParallelService_OeBB_wrong.xml:5]: done

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml (hunk 5 above)
Comment (ue71603) [inline examples/functions/trip/TR_ParallelService_OeBB_wrong.xml:146]:
```suggestion
										<!-- the PublishedService changes during the Leg from NJ490 to NJ590, therefore having this PublishedServiceName and the train number
```

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+											<Text xml:lang="de">Zug</Text>
+										</Name>
+										<ShortName>
+											<Text xml:lang="de">NJ</Text>
+										</ShortName>
+									</Mode>
+									<PublishedServiceName>
+										<Text xml:lang="de">NJ 490</Text>
+										<!-- the PublishedService changes in the middle, therefore having this PublishedServiceName and the train number
+										for the whole Leg would be misleading. See correct example -->
```
Comment (ue71603) [inline examples/functions/trip/TR_ParallelService_OeBB_wrong.xml:147]:
```suggestion
										for the whole Leg would be misleading. See correct example in the folder. -->
```

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml (hunk 6 above)
Comment (ue71603) [inline examples/functions/trip/TR_ParallelService_OeBB_wrong.xml:219]:
```suggestion
									<!-- the German part of the train to Amsterdam. It is connected only up to Nürberg with the relevant vehicle of this TimedLeg. As it has its own, separated JOURNEY afterwards, it is modeled in ParallelService only to Nürnberg. -->
```

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml (hunk 4 above)
Comment (ue71603) [reply examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml]: changed

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml (hunk 5 above)
Comment (ue71603) [reply examples/functions/trip/TR_ParallelService_OeBB_wrong.xml:148]: changed

File: examples/functions/trip/TR_ParallelService_OeBB_wrong.xml (hunk 6 above)
Comment (ue71603) [reply examples/functions/trip/TR_ParallelService_OeBB_wrong.xml]: changed.

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml (hunk 3 above)
Comment (ue71603) [reply examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml:26]: @herlitze  = zero transfers.

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+				<CalcTime>762</CalcTime>
+				<TripResult>
+					<Id>ID-48D1A6A7-2659-4433-9795-F0A0840A4938</Id>
+					<Trip>
+						<Id>ID-48D1A6A7-2659-4433-9795-F0A0840A4938</Id>
+						<Duration>PT48M</Duration>
+						<StartTime>2023-08-15T18:08:00Z</StartTime>
+						<EndTime>2023-08-15T18:56:00Z</EndTime>
+						<Transfers>1</Transfers>
+						<!-- is one even when it is a remain seated -->
```
Comment (ue71603) [inline examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml:26]:
```suggestion
						<Transfers>0</Transfers>
						<!-- TransferLegs with remainInVehicle are NOT counted.  -->
```

File: examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml (hunk 3 above)
Comment (ue71603) [reply examples/functions/trip/TR_ParallelService_OeBB_right_with_remain_in_vehicle.xml]: changed

**Review by sgrossberndt — APPROVED** [review]

**Pull request merged**